### ISSUE TITLE
Add support for !~ to Style/SelectByRegexp

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#11219]
Currently, `Style/SelectByRegexp` recognizes `=~` but not `!~`:

```ruby
list.select { |n| n =~ /good/ } # cop recommends list.grep(/good/) ✔️
list.select { |n| n !~ /bad/ }  # cop ignores 😔
```

Ideally, it would recommend `list.grep_v(/bad/)`
